### PR TITLE
Qt: Add vertical spacer to peer detail widget

### PR DESCRIPTION
**Pull request by JosuGZ**
Before:

image

After:

image

### CODE REVIEW

**Comment by practicalswift** [comment]
Concept ACK: looks much better!

@JosuGZ, welcome as a contributor! Hope to see more contributions from you! :-)

**Comment by JosuGZ** [comment]
> Concept ACK: looks much better!
> 
> @JosuGZ, welcome as a contributor! Hope to see more contributions from you! :-)

Thanks! I have no clue about why that `rpc_psbt.py` test fails though.

**Comment by practicalswift** [comment]
@joshrabinowitz That is just a random unrelated failure :-)

**Review by kristapsk — APPROVED** [review]
ACK 36b0a2f2a6b49008c4f37866a9e3ab702eb34eda (tested with Qt 5.11.3 under Linux/Xfce4)

File: src/qt/forms/debugwindow.ui [hunk 1]
```diff
@@ -1480,6 +1480,19 @@
                  </property>
                 </widget>
                </item>
+               <item row="18" column="0">
+                <spacer name="verticalSpacer_3">
+                 <property name="orientation">
+                  <enum>Qt::Vertical</enum>
+                 </property>
+                 <property name="sizeHint" stdset="0">
```
Comment (promag) [inline src/qt/forms/debugwindow.ui]: Property could be removed.

**Review by promag — COMMENTED** [review]
Tested ACK https://github.com/bitcoin/bitcoin/pull/16090/commits/36b0a2f2a6b49008c4f37866a9e3ab702eb34eda on macos 10.14.3. Resizing the window works as expected.

**Comment by fanquake** [comment]
utACK https://github.com/bitcoin/bitcoin/pull/16090/commits/36b0a2f2a6b49008c4f37866a9e3ab702eb34eda

This change doesn't seem to make any difference on macOS (`10.14.x`), as the `before` behavior isn't present.

@JosuGZ What OS are you seeing this issue on?

**Comment by promag** [comment]
@fanquake I'm on a macos and it also stretches the details (macos 10.14.3, Qt 5.12.2).

**Comment by kristapsk** [comment]
@fanquake , I have the "before" behaviour also on Linux/Xfce4, Qt 5.11.3.

**Comment by JosuGZ** [comment]
@fanquake 
> @JosuGZ What OS are you seeing this issue on?

Linux Mint 19.1 Tessa

It usually also happens when I do this kind of layouts on Windows.

**Comment by fanquake** [comment]
@promag @kristapsk @JosuGZ Thanks for clarifying.

File: src/qt/forms/debugwindow.ui (hunk 1 above)
Comment (JosuGZ) [reply src/qt/forms/debugwindow.ui]: It seems indeed unneeded, I think it is how Qt creator creates spacers though. Change pushed.

**Comment by hebasto** [comment]
tACK 36b0a2f2a6b49008c4f37866a9e3ab702eb34eda on Linux Mint 19.1
Could you squash your commits?

The different behavior (some can see 'before', others not) can be caused by [QGridLayout::setRowStretch](https://doc.qt.io/qt-5/qgridlayout.html#setRowStretch):
> If the stretch factor is 0 and no other row in this table can grow at all, the row _may_ still grow.

File: src/qt/forms/debugwindow.ui (hunk 1 above)
Comment (JosuGZ) [reply src/qt/forms/debugwindow.ui]: Hmm, I'm noticing other spacers have the property set, like the one in the information tab on the same file.

**Comment by JosuGZ** [comment]
@hebasto I have left the property for coherency with the rest of the file and with Qt Creator.

**Comment by fanquake** [comment]
re-utACK https://github.com/bitcoin/bitcoin/commit/36b0a2f2a6b49008c4f37866a9e3ab702eb34eda

Could have used a commit message like `gui: add vertical spacer to peers window`.

**Comment by JosuGZ** [comment]
True. Shouldn't appear in the the merge commit anyway?

**Pull request merged**